OpenRA.Mods.Common/Traits/Sound/ActorLostNotification.cs

```diff
@@ -20,21 +20,29 @@ class ActorLostNotificationInfo : ConditionalTraitInfo
 
 		public readonly bool NotifyAll = false;
 
-		public override object Create(ActorInitializer init) { return new ActorLostNotification(this); }
+		public override object Create(ActorInitializer init) { return new ActorLostNotification(this, init.Self); }
 	}
 
 	class ActorLostNotification : ConditionalTrait<ActorLostNotificationInfo>, INotifyKilled
 	{
-		public ActorLostNotification(ActorLostNotificationInfo info)
-			: base(info) { }
+		readonly World world;
+		readonly Player player;
+		readonly FactionInfo faction;
+
+		public ActorLostNotification(ActorLostNotificationInfo info, Actor self)
+			: base(info)
+		{
+			world = self.World;
+			player = Info.NotifyAll ? world.LocalPlayer : self.Owner;
```

Comment:
What if the actor is captured and therefore changes owner midgame?

Reply:
Yeah, this change is misleading. This trait is not always the source so this won't help at all.